lib/sunbeam-ws.js

```diff
@@ -226,17 +231,20 @@ class MandelbrotEosfinex extends EventEmitter {
     const {
       keys,
       client,
-      scatter
+      ual
     } = auth
 
     if (keys && client) {
       this.conf.eos.auth.keys = keys
-      this.client = client
-      this.signer = new SignHelper(this.client, this.conf)
+      this.setClient(client, this.conf)
       return
     }
 
-    this.scatter = new SunbeamScatter(scatter)
+    if (client) {
+      this.setClient(client)
+    }
+
+    this.wallet = new SunbeamUAL(ual)
```

Comment:
are we completely switching to `ual` in 100% cases?

i see scatter can be initialized in the constructor but `ual` can not, is it intended to be so?

Reply:
we are switching to `ual`, but not yet in sunbeam :)
passing it into the constructor is only used by sunbeam example code. We cannot do this in the UI, because the sunbeam instance is created long before they auth. Because it also handles public requests.
We use `setAuth` in the UI.

the problem is, that `ual-scatter` does not work with cosign, so we have our own impl. of it in the UI
and `ual-anchor` only works in there is a DOM available.

I will port the UI impl. of ual-scatter into sunbeam, so we can use that for example/test. And then remove all the native scatter impl.

But i would like to see it work, before we strip it all out. The current examples in sunbeam still work with the native Scatter impl. inside sunbeam, so i left it there for now.

Reply:
am replacing it now, hang on

Reply:
should be fixed now